lib/rubocop/cop/lint/useless_access_modifier.rb

```diff
@@ -145,6 +151,16 @@ def on_block(node)
         alias on_numblock on_block
         alias on_itblock on_block
 
+        def on_begin(node)
+          return if node.parent
+
+          node.child_nodes.each do |child|
+            next unless child.send_type? && access_modifier?(child)
+
+            check_send_node(child, child.method_name, child)
```

Comment:
This is a bit hacky, but I was aiming to reuse as much of the existing code as possible. Namely, this part: https://github.com/rubocop/rubocop/blob/e236de8f52b72d73418de11e4eea6dd03d9c6a6d/lib/rubocop/cop/lint/useless_access_modifier.rb#L199-L200

Let me know if I should refactor this somehow or add an explanatory comment.